pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1110,106 +1282,103 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListenerForPortOrUDS(l
-
-		compareWithExisting:
-			for _, existingFilterChain := range currentListenerEntry.listener.FilterChains {
-				if existingFilterChain.FilterChainMatch == nil {
-					// This is a catch all filter chain.
-					// We can only merge with a non-catch all filter chain
-					// Else mark it as conflict
-					if incomingFilterChain.FilterChainMatch == nil {
-						// NOTE: While pluginParams.Service can be nil,
-						// this code cannot be reached if Service is nil because a pluginParams.Service can be nil only
-						// for user defined Egress listeners with ports. And these should occur in the API before
-						// the wildcard egress listener. the check for the "locked" bit will eliminate the collision.
-						// User is also not allowed to add duplicate ports in the egress listener
-						var newHostname host.Name
-						if pluginParams.Service != nil {
-							newHostname = pluginParams.Service.Hostname
-						} else {
-							// user defined outbound listener via sidecar API
-							newHostname = "sidecar-config-egress-tcp-listener"
-						}
-
-						conflictFound = true
-						outboundListenerConflict{
-							metric:          model.ProxyStatusConflictOutboundListenerTCPOverTCP,
-							node:            pluginParams.Node,
-							listenerName:    listenerMapKey,
-							currentServices: currentListenerEntry.services,
-							currentProtocol: currentListenerEntry.servicePort.Protocol,
-							newHostname:     newHostname,
-							newProtocol:     pluginParams.Port.Protocol,
-						}.addMetric(pluginParams.Push)
-						break compareWithExisting
-					} else {
-						continue
-					}
-				}
-				if incomingFilterChain.FilterChainMatch == nil {
-					continue
-				}
-
-				// We have two non-catch all filter chains. Check for duplicates
-				if reflect.DeepEqual(*existingFilterChain.FilterChainMatch, *incomingFilterChain.FilterChainMatch) {
-					var newHostname host.Name
-					if pluginParams.Service != nil {
-						newHostname = pluginParams.Service.Hostname
-					} else {
-						// user defined outbound listener via sidecar API
-						newHostname = "sidecar-config-egress-tcp-listener"
-					}
-
-					conflictFound = true
-					outboundListenerConflict{
-						metric:          model.ProxyStatusConflictOutboundListenerTCPOverTCP,
-						node:            pluginParams.Node,
-						listenerName:    listenerMapKey,
-						currentServices: currentListenerEntry.services,
-						currentProtocol: currentListenerEntry.servicePort.Protocol,
-						newHostname:     newHostname,
-						newProtocol:     pluginParams.Port.Protocol,
-					}.addMetric(pluginParams.Push)
-					break compareWithExisting
-				}
-			}
+	// If there is a TCP listener on well known port, cannot add any http filter chain
+	// with the inspector as it will break for server-first protocols. Similarly,
+	// if there was a HTTP listener on well known port, cannot add a tcp listener
+	// with the inspector as inspector breaks all server-first protocols.
+	if currentListenerEntry != nil &&
```

Comment:
does this code need to be flag protected?

Reply:
nope. this has been here for a while. just code moves